h2o-r/h2o-package/R/communication.R

```diff
@@ -583,17 +582,56 @@ print.H2OTable <- function(x, header=TRUE, ...) {
   }
 
   rawREST <- ""
-
   if( !is.null(timeout) ){
     rawREST <- .h2o.doSafeREST(h2oRestApiVersion = h2oRestApiVersion, urlSuffix = page, parms = .params, method = method, timeout = timeout)
   }else if(autoML == TRUE){
     rawREST <- .h2o.doSafeREST(h2oRestApiVersion = h2oRestApiVersion, urlSuffix = page, parms = .params, method = method,autoML=autoML)
   }else{
     rawREST <- .h2o.doSafeREST(h2oRestApiVersion = h2oRestApiVersion, urlSuffix = page, parms = .params, method = method)
   }
+  if(raw) { 
+    rawREST
+  }
+  else { 	  
+    jsonREST <- .h2o.fromJSON(jsonlite::fromJSON(rawREST,simplifyDataFrame=FALSE))
+    if(grepl("seed", rawREST)){
+        jsonREST <- .h2o.__parseSeedCorrectly(rawREST, jsonREST)
+    }
+    jsonREST  
+  }
+}
 
-  if( raw ) rawREST
-  else      .h2o.fromJSON(jsonlite::fromJSON(rawREST,simplifyDataFrame=FALSE))
+# The problem is when users create a model without seed in R, they are not able to get correct seed autogenerated from Java, 
+# because it is in format, which can't be used in R without loss. 
+# For example h2o generates -8664354335142703762 but in R user the number changes to -8664354335142704128,
+# so this number should be read and pass as string in R. 
+# It is strongly API related, if someone change the structure of the REST, it can affect this method.
+# PUBDEV-6262 Not able to reproduce GBM in R if seed not set
+.h2o.__parseSeedCorrectly <- function(jsonRaw, jsonParsed){
+    expr = regexpr("\"seed[^}]*\"actual_value\":-?([0-9])+", jsonRaw, perl=TRUE)
+    seedString = gsub("\"seed[^}]*\"actual_value\":", "", regmatches(jsonRaw, expr), perl=TRUE)
+    if(jsonParsed$"__meta"$schema_name == "ModelBuildersV3"){
+        modelId = "model_builders"
+    } else if(jsonParsed$"__meta"$schema_name == "ModelsV3"){
+        modelId = "models"
+    } else{
+        # FIXME possibly different schema with seed?
+        return(jsonParsed)
+    }
+    parameters = jsonParsed[[modelId]][[1]]$parameters
+    for (i in 1:length(parameters)){
+        if(parameters[[i]]$name=="seed"){
+            seedIndex = i
+            break
+        }
+    }
+    seedParsed = parameters[[seedIndex]]$actual_value
+    # test if seed is corrupted by R, if yes convert it to string 
+    if (is.numeric(seedParsed) && toString(seedParsed) != seedString){
```

Comment:
The drawback of this approach is that people will sometimes get a number back and sometimes a string. That can be surprising and perhaps also cause issues.

Maybe we should keep seed as String always. WDYT?

The issue is also not limited to seed, any integer number can have this issue.

Reply:
So input can be long and string too, and output always string? Only in R?  I think it is also confusing, however still better than now.